activerecord/test/cases/relation_test.rb

```diff
@@ -294,6 +294,16 @@ def test_relation_merging_with_joins_as_join_dependency_pick_proper_parent
       assert_equal 3, relation.where(id: post.id).pluck(:id).size
     end
 
+    def test_relation_merging_with_joins_and_unscope_on_merged_relation
+      post1 = Post.create!(title: "foo", body: "foo body")
+      post1.comments.create!(body: "foo comment 1")
+      post2 = Post.create!(title: "bar", body: "bar body")
+      post2.comments.create!(body: "bar comment 1")
+
+      relation = Post.where(body: "foo body").joins(:comments).merge(Comment.where("1=0").unscope(:where))
```

Comment:
How should `Post.where(body: "foo body").joins(:comments).merge(Comment.where("1=0").unscope(where: :"posts.body"))` behave?